gui/packages/ubuntupro/lib/pages/landscape/landscape_model.dart

```diff
@@ -236,24 +248,38 @@ class LandscapeSelfHostedConfig extends LandscapeConfig {
       return true;
     }
 
-    final fileStat = File(path).statSync();
+    final file = File(path);
+    final fileStat = file.statSync();
+
     if (fileStat.type == FileSystemEntityType.notFound) {
       _fileError = FileError.notFound;
     } else if (fileStat.type == FileSystemEntityType.directory) {
       _fileError = FileError.dir;
     } else if (fileStat.size == 0) {
       _fileError = FileError.emptyFile;
+    } else if (validCertExtensions.every((e) => !file.path.endsWith(e))) {
+      _fileError = FileError.invalidFormat;
     } else {
-      _fileError = FileError.none;
+      final content = file.readAsBytesSync();
+      try {
+        X509.fromDer(content);
+        _fileError = FileError.none;
+        // Various exception or errors can occur here when attempting a parse
+        // ignore: avoid_catches_without_on_clauses
+      } catch (_) {
```

Comment:
This leads to `analyze` errors despite being ignored. This is needed since both Exceptions during decoding happen as well as various Errors, both meaning that the cert is invalid (the tests now include valid and invalid binary files for testing this purpose). Maybe there's a better way to do this other than error catching but I don't believe anything is built into X509 to check for cert validity.

Reply:
You can always catch by the base class (`Exception` in this case) as you did below. I quickly look at the underlying parsing routines and they seem to only throw Exception-derived objects. 

If you've seen `Error`-derived objects then we should have a pairing session to look into it more carefully. `Error` objects are by definition to not be caught.

Reply:
I'd also move this part into its own function because it's not the prettiest code we've ever written :)

Reply:
This is an Error that can occur if I try to load the `not_a_cert.pem` from `testdata/`:

```
[+12597 ms] [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: RangeError:
Value not in range: 106
[        ] #0      _rangeCheck (dart:typed_data-patch/typed_data_patch.dart:5119:5)
[        ] #1      _ByteBuffer.asUint8List (dart:typed_data-patch/typed_data_patch.dart:1928:5)
[        ] #2      new Uint8List.view (dart:typed_data:903:19)
[        ] #3      ASN1Parser.nextObject (package:pointycastle/asn1/asn1_parser.dart:74:30)
[        ] #4      new X509.fromDer (package:pkcs7/src/x509.dart:25:25)
[        ] #5      LandscapeSelfHostedConfig._validatePath
(package:ubuntupro/pages/landscape/landscape_model.dart:264:12)
[        ] #6      LandscapeSelfHostedConfig.sslKeyPath=
(package:ubuntupro/pages/landscape/landscape_model.dart:283:9)
[        ] #7      LandscapeModel.setSslKeyPath
(package:ubuntupro/pages/landscape/landscape_model.dart:102:16)
[        ] #8      _FilePickerFieldState.build.<anonymous closure>
(package:ubuntupro/pages/landscape/landscape_page.dart:477:31)
[        ] <asynchronous suspension>
```
*(This is with the code slightly modified to actually output errors so the line numbers aren't accurate. The offending call is `X509.fromDer(content);` in `_validatePath`)*

Most other files I've tried will indeed lead to Exceptions instead of Errors, but not this one.

Reply:
Ok. I see what's going on. Your catch-all approach makes sense. Let's just move this piece into its own function and voilá!